### PR TITLE
[pod repo push] Discover and push JSON specs while pushing from a directory

### LINKED ISSUE #3494: [Command Repo Push] No globbing for JSON podspecs
The repo push command doesn't glob for the newer `podspec.json` format, but it is also acceptable as parameter. This is misleading in combination with the error message, which only covers the case of pushing all podspecs in the current dir.

```shell
$ pod repo push my-repo MyPod.podspec
[!] Couldn't find any .podspec file in current directory
```

I'd propose the error message should be the following, if a file argument was specified by the user:
```
[!] Couldn't find file `MyPod.podspec.json` in current directory
```

If no podspec was specified, the error message should be:
```
[!] Couldn't find any podspec file (*.podspec{,.json}) in current directory
```

### PR DESCRIPTION
**Pull request by kylef**
Closes #3494.

### CODE REVIEW

File: spec/functional/command/repo/push_spec.rb [hunk 1]
```diff
@@ -38,6 +44,16 @@ module Pod
       end
     end
 
+    it 'finds JSON podspecs' do
+      repo_make('test_repo')
+
+      Dir.chdir(temporary_directory) do
+        File.open('JSON.podspec.json',  'w') { |f| f.write('{}') }
+        cmd = command('repo', 'push', 'test_repo')
+        cmd.send(:podspec_files).should == [Pathname('JSON.podspec.json')]
```
Comment (kylef) [inline spec/functional/command/repo/push_spec.rb]: If anyone has any suggestions on how to easily test this without accessing the private method I'm open for ideas.

**Comment by segiddins** [comment]
:+1:

**Comment by mrackwitz** [comment]
:+1:

File: spec/functional/command/repo/push_spec.rb (hunk 1 above)
Comment (segiddins) [inline spec/functional/command/repo/push_spec.rb]: this is perfect the way it is

**Pull request merged**